sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetLogRedirector.java

```diff
@@ -25,11 +25,11 @@
 
 // Redirects the JUL logging for parquet-mr versions <= 1.8 to SLF4J logging using
 // SLF4JBridgeHandler. Parquet-mr versions >= 1.9 use SLF4J directly
-final class ParquetLogRedirector implements Serializable {
+public final class ParquetLogRedirector implements Serializable {
```

Comment:
Spark uses Parquet >= 1.9. Is this still needed? Why was it made public?

Reply:
> Spark uses Parquet >= 1.9. Is this still needed?

I am not sure about this. I think we can resolve this in another Jira/PR.

> Why was it made public?

We need to make it public so that ParquetWriteBuilder can access it. As per the discussion in https://issues.apache.org/jira/browse/SPARK-16964, I think it is fine to do this in the `sql.execution` package

Reply:
> I am not sure about this. I think we can resolve this in another Jira/PR.

Do you plan to follow this up with a PR to remove it? If not, then we should decide here and avoid copying it to v2.

Reply:
OK, I think we can remove `ParquetLogRedirector.INSTANCE` in V2 to make it simple. But users might still hit the issue in https://github.com/apache/spark/pull/15538 if they tried to read files written using Parquet-mr 1.6.0.

Reply:
Why would the version a file was written with matter? Does Spark use 1.6.0 to read those files?

Reply:
Sorry, I was following the description in #15538 . I can still see 
```
<hive.parquet.version>1.6.0</hive.parquet.version>
```
in pom.xml.  I will do more investigation for this.

Reply:
@mallman @srowen @wangyum Do you know whether we can remove the Parquet log output redirection here?

Reply:
Hm, I think we can stop supporting Parquet 1.6 already. I'm not sure why this still references 1.6.0 vs `parquet.version`. I presume we get away with it because it's the com.twitter parquet and it's shaded or something? maybe it's required for Hadoop 2.x support.

Reply:
@srowen Thanks. I think we can skip the Parquet log output redirection in Parquet V2. Since Spark is using Parquet 1.10, and it uses native Parquet file format by default when reads/writes with Hive.

Reply:
> Spark uses Parquet >= 1.9. Is this still needed? Why was it made public?

I had to dig into the "archive" to remember why we did this. This whole rigmarole is explained in b533fa2.

> Why would the version a file was written with matter? Does Spark use 1.6.0 to read those files?

The version of the Parquet writer matters because the Parquet reader was complaining about invalid metadata written by the Parquet 1.6 writer. Another, prior commit, 6ce1b67, had the side-effect of breaking the workaround that allowed Spark users to specify logging configuration for Parquet through Spark's logging configuration. We punted a fix for this regression to b533fa2.

According to the top level `pom.xml` at commit b533fa2, Spark was using Parquet 1.8.1. Based on that and the comment in this code

```
// Redirects the JUL logging for parquet-mr versions <= 1.8 to SLF4J logging using
// SLF4JBridgeHandler. Parquet-mr versions >= 1.9 use SLF4J directly
```

it looks like this workaround was necessary at the time. Given that Spark now uses Parquet 1.10.1, it appears this workaround is no longer necessary—this code can be deleted.

N.B. that according to the commit message of b533fa2, this workaround was only applicable for the case of converted partitioned Hive tables. Therefore, I presume that the version of the Hive Parquet library Spark uses is irrelevant.

All that said, there are no unit tests to guard against regression. To be safe I would recommend going through the manual, 4-step test procedure I outlined in b533fa2 if this workaround is removed.

Hope this helps.

Reply:
@mallman Would you mind checking it manually and creating a PR to remove the log redirection? I think you are the best person. Thanks in advance!

Reply:
Hi @gengliangwang. I won't be able to help you with this, no. I don't have access to the environment or data necessary for testing, and I cannot make time for the PR. Sorry to disappoint.

Regardless, I suggest just deleting this code as part of this PR. Given that Spark is now using Parquet 1.10.1, and the fact that Parquet 1.9+ logs with SLF4J directly, I think deleting this code is low-enough risk.

Reply:
@mallman I see.  Appreciate your help!

Reply:
`hive.parquet.version` sets the version of Parquet used by Hive, when Spark is wrapping a table and using Hive readers. This doesn't affect either the v1 Parquet data source or the v2 source. This may still be necessary for Hive reads, but I don't think it needs to be carried forward to v2.

Reply:
I have removed it in Parquet V2. Mark this thread as resolved.